In [0]:
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error

In [0]:
# load data
train_data = pd.read_csv("data/train.csv")
test_data = pd.read_csv("data/test.csv")

In [0]:
# feature selection
features = ["Pclass", "Sex", "SibSp", "Parch"]

In [0]:
X = pd.get_dummies(train_data[features])
y = train_data['Survived']

In [0]:
# split data to train set and validation set
train_X, val_X, train_y, val_y = train_test_split(X, y, random_state = 0)

In [33]:
# Random forest prediction
forest_model = RandomForestClassifier(n_estimators=100)
forest_model.fit(train_X, train_y)
forest_val_predictions = forest_model.predict(val_X)
print('MAE of Random Forests: %f' % (mean_absolute_error(val_y, forest_val_predictions)))

MAE of Random Forests: 0.210762


In [34]:
# Decision Tree prediction
tree_model = DecisionTreeRegressor(random_state=1)
tree_model.fit(train_X, train_y)
tree_val_predictions = tree_model.predict(val_X)
print('MAE of Decision Tree: %f' % (mean_absolute_error(val_y, tree_val_predictions)))

MAE of Decision Tree: 0.275228


In [59]:
# generate CSV
output_1 = pd.DataFrame({'PassengerId': test_data.PassengerId})
print(output_1)
output_1.to_csv('my_submission_1.csv', index=False)

     PassengerId
0            892
1            893
2            894
3            895
4            896
..           ...
413         1305
414         1306
415         1307
416         1308
417         1309

[418 rows x 1 columns]


In [60]:
# generate CSV
output_2 = pd.DataFrame({'Survived': forest_val_predictions})
print(output_2)
output_2.to_csv('my_submission_2.csv', index=False)


     Survived
0           0
1           0
2           0
3           1
4           1
..        ...
218         0
219         1
220         0
221         1
222         0

[223 rows x 1 columns]


**Code After Solving array length 223 does not match index length 418!!!!!!!!!!!!**

In [0]:
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error

In [65]:
test_data = pd.read_csv("data/test.csv")
test_data.head()

PassengerId  Pclass  ... Cabin Embarked
0          892       3  ...   NaN        Q
1          893       3  ...   NaN        S
2          894       2  ...   NaN        Q
3          895       3  ...   NaN        S
4          896       3  ...   NaN        S

[5 rows x 11 columns]

In [66]:
women = train_data.loc[train_data.Sex == 'female']["Survived"]
rate_women = sum(women)/len(women)

print("% of women who survived", rate_women)

% of women who survived 0.7420382165605095


In [67]:
men = train_data.loc[train_data.Sex == 'male']["Survived"]
rate_men = sum(men)/len(men)

print("% of men who survived:", rate_men)

% of men who survived: 0.18890814558058924


In [69]:
from sklearn.ensemble import RandomForestClassifier

y = train_data["Survived"]

features = ["Pclass", "Sex", "SibSp", "Parch"]
X = pd.get_dummies(train_data[features])
X_test = pd.get_dummies(test_data[features])

model = RandomForestClassifier(n_estimators=100, max_depth=5, random_state=1)
model.fit(X, y)
predictions = model.predict(X_test)

output = pd.DataFrame({'PassengerId': test_data.PassengerId, 'Survived': predictions})
print(output)
output.to_csv('my_submission.csv', index=False)
print("Your submission was successfully saved!")

     PassengerId  Survived
0            892         0
1            893         1
2            894         0
3            895         0
4            896         1
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]
Your submission was successfully saved!
